pkg/codegen/importer/hcl2.go

```diff
@@ -425,7 +426,68 @@ func generateValue(typ schema.Type, value resource.PropertyValue) (model.Express
 					})
 				}
 			}
-		} else {
+		case *schema.UnionType:
+			discriminatorPropertyName := objectType.Discriminator
+			selectElementType := func() (*schema.ObjectType, bool) {
+				discriminatorValue, ok := obj[resource.PropertyKey(discriminatorPropertyName)]
+				if !ok {
+					// discriminator property is not present, so we can't select a type
```

Comment:
I _think_ we need to fallback to the untyped map cases if we fail to work out the union case type correctly? @iwahbe or @justinvp might be able to confirm that, I'm not totally sure what the constraints and expectations of this method are.